OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignalCacheCleaner.java

```diff
@@ -4,69 +4,219 @@
+    private static void cleanCachedNotifications(SQLiteDatabase writableDb) {
+        String whereStr = NotificationTable.COLUMN_NAME_CREATED_TIME + " < ?";
+
+        String sevenDaysAgoInSeconds = String.valueOf((System.currentTimeMillis() / 1_000L) - NOTIFICATION_CACHE_DATA_LIFETIME);
+        String[] whereArgs = new String[]{ sevenDaysAgoInSeconds };
+
+        writableDb.delete(
+                NotificationTable.TABLE_NAME,
+                whereStr,
+                whereArgs);
     }
 
     /**
-     * Deletes any notifications with created timestamps older than 7 days
+     * Deletes cached unique outcome notifications whose ids do not exist inside of the NotificationTable.TABLE_NAME
+     * <br/><br/>
+     * Note: This should only ever be called by {@link OneSignalCacheCleaner#cleanNotificationCache(SQLiteDatabase)}
+     * <br/><br/>
+     * @see OneSignalCacheCleaner#cleanNotificationCache(SQLiteDatabase)
      */
-    private static void cleanOldNotificationData(SQLiteDatabase writableDb) {
-        writableDb.delete(NotificationTable.TABLE_NAME,
-                NotificationTable.COLUMN_NAME_CREATED_TIME + " < " + ((System.currentTimeMillis() / 1_000L) - 604_800L),
+    private static void cleanCachedUniqueOutcomeEventNotifications(SQLiteDatabase writableDb) {
+        String whereStr = "NOT EXISTS(" +
+                "SELECT NULL FROM " + NotificationTable.TABLE_NAME + " n " +
+                "WHERE" + " n." + NotificationTable.COLUMN_NAME_NOTIFICATION_ID + " = " + CachedUniqueOutcomeNotificationTable.COLUMN_NAME_NOTIFICATION_ID + ")";
+
+        writableDb.delete(
+                CachedUniqueOutcomeNotificationTable.TABLE_NAME,
+                whereStr,
                 null);
     }
 
     /**
-     * Deletes any notifications whose ids do not exist inside of the NotificationTable.TABLE_NAME
+     * Deletes old IAM SharedPreference dismissed and impressioned message ids as well as clicked click ids
+     * <br/><br/>
+     * Note: This should only ever be called by {@link OneSignalCacheCleaner#cleanCachedInAppMessages(SQLiteDatabase)}
+     * <br/><br/>
+     * @see OneSignalCacheCleaner#cleanCachedInAppMessages(SQLiteDatabase)
      */
-    static void cleanOldUniqueOutcomeEventNotificationsCache(SQLiteDatabase writableDb) {
-        writableDb.delete(CachedUniqueOutcomeNotificationTable.TABLE_NAME,
-                "NOT EXISTS(SELECT NULL FROM " + NotificationTable.TABLE_NAME +
-                        " n WHERE" +
-                        " n." + NotificationTable.COLUMN_NAME_NOTIFICATION_ID  + " = " + CachedUniqueOutcomeNotificationTable.COLUMN_NAME_NOTIFICATION_ID + ")",
-                null);
+    private static void cleanCachedSharedPreferenceIamData(Set<String> oldMessageIds, Set<String> oldClickedClickIds) {
+        // IAMs without redisplay on with pile up and we need to clean these for dismissing, impressions, and clicks
+        if (oldMessageIds != null && oldMessageIds.size() > 0) {
```

Comment:
This would be cleaner with 2 methods instead of long if blocks.

Reply:
On it!